TcpIntegrationSpec: should properly handle connection abort from client side

(there are a couple of other `TcpIntegrationSpec` failures that could be duplicates, but let's confirm that first)

https://github.com/akka/akka/runs/3278202293?check_suite_focus=true

```
[info] - should properly handle connection abort from client side *** FAILED *** (3 seconds, 46 milliseconds)
[info]   java.lang.AssertionError: assertion failed: timeout (3 seconds) during expectMsgClass waiting for class akka.io.Tcp$ErrorClosed
[info]   at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
[info]   at akka.testkit.TestKitBase.expectMsgClass_internal(TestKit.scala:571)
[info]   at akka.testkit.TestKitBase.expectMsgType(TestKit.scala:543)
[info]   at akka.testkit.TestKitBase.expectMsgType$(TestKit.scala:159)
[info]   at akka.testkit.TestKit.expectMsgType(TestKit.scala:973)
[info]   at akka.io.TcpIntegrationSpec$$anon$2.<init>(TcpIntegrationSpec.scala:49)
[info]   at akka.io.TcpIntegrationSpec.f$proxy3$1(TcpIntegrationSpec.scala:52)
[info]   at akka.io.TcpIntegrationSpec.$init$$$anonfun$4$$anonfun$3(TcpIntegrationSpec.scala:45)
```

--- a/akka-actor/src/main/scala/akka/io/TcpConnection.scala
+++ b/akka-actor/src/main/scala/akka/io/TcpConnection.scala
@@ -74,11 +74,12 @@ private[io] abstract class TcpConnection(val tcp: TcpExt, val channel: SocketCha
 
       val info = ConnectionInfo(registration, handler, keepOpenOnPeerClosed, useResumeWriting)
 
+      context.setReceiveTimeout(Duration.Undefined)
+      context.become(connected(info))
+
       // if we are in push mode or already have resumed reading in pullMode while waiting for Register
       // then register OP_READ interest
       if (!pullMode || (/*pullMode && */ !readingSuspended)) resumeReading(info, None)
-      context.setReceiveTimeout(Duration.Undefined)
-      context.become(connected(info))
 
     case ResumeReading =>
       readingSuspended = false

